src/test/groovy/graphql/schema/idl/TypeDefinitionRegistryTest.groovy

```diff
@@ -985,6 +988,215 @@ class TypeDefinitionRegistryTest extends Specification {
         "Bar"           | _
     }
 
+    def "can maintain parsed order"() {
+        def sdl = '''
+            "the schema"
+            schema {
+                query : Q
+            }
+            
+            "the query type"
+            type Q {
+                field( arg : String! = "default") : FieldType @deprecated(reason : "no good")
```

Comment:
So, the comparator doesn't sort fields, but `getInOrder` does it, correct? Perhaps we could add some more fields in this test and assert the order is preserved?

Reply:
getInOrder gives back the SDLDefinition in the order  of parsing

The comparator allows a runtime set of types to be sorted back on the parsed order (eg not SDLDefintions AST things but runtime type)